### PR TITLE
Followup regarding subxt test

### PR DESCRIPTION
**Pull request by xermicus**
* Do not rely on a CI node to be able to compile the tests
* Compile the tests in parallel
* Add a short README  hinting about how to maintain this
* Set me as the codeowner

@extraymond if you see this and wondering why providing the metadata directly (again) instead of fetching from the node. I did not think about that if we fetch it directly, anyone clonening the Solang repo will also have to have our CI node running, otherwise compilation fails. This is a bit awkward so we decided to change it again.

### CODE REVIEW

File: integration/subxt-tests/src/lib.rs [hunk 1]
```diff
@@ -28,7 +31,7 @@ mod cases;
 
 // metadata file obtained from the latest substrate-contracts-node
 #[subxt::subxt(
-    runtime_metadata_url = "ws://localhost:9944",
+    runtime_metadata_path = "./metadata.scale",
```
Comment (seanyoung) [inline integration/subxt-tests/src/lib.rs:34]: nice

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -373,7 +373,7 @@ jobs:
         toolchain: stable
         default: true
         override: true   
-    - run: solang compile --target substrate ../substrate/*.sol ../substrate/test/*.sol -o ./contracts/
+    - run: 'parallel solang compile -v --target substrate --release -o ./contracts/ ::: ../substrate/*.sol ../substrate/test/*.sol'
```
Comment (seanyoung) [inline .github/workflows/test.yml:376]: This would be nice to have in solang

**Review by seanyoung — APPROVED** [review]

File: .github/workflows/test.yml (hunk 2 above)
Comment (xermicus) [reply .github/workflows/test.yml:376]: Yeah, definitively. #1382

File: integration/subxt-tests/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+## How to execute the tests
+
+The test cases requires compiled versions of the contracts from the `substrate` integration test suite inside the `./outputs` dir. To compile everything, run:
+
+```bash
+parallel solang compile -v --target substrate --release -o ./contracts/ ::: ../substrate/*.sol ../substrate/test/*.sol
+```
+
+Make sure start a [solang-substrate-ci node](https://github.com/hyperledger/solang-substrate-ci) or a [substrate-contracts-node](https://github.com/paritytech/substrate-contracts-node) on the test host.
```
Comment (LucasSte) [inline integration/subxt-tests/README.md:13]:
```suggestion
Make sure to start a [solang-substrate-ci node](https://github.com/hyperledger/solang-substrate-ci) or a [substrate-contracts-node](https://github.com/paritytech/substrate-contracts-node) on the test host.
```

**Review by LucasSte — APPROVED** [review]

**Pull request merged**